File: backend/server/src/sparkClient.js
import axios from 'axios'
import http from 'node:http'
import https from 'node:https'

const SPARK_BASE_URL = (process.env.SPARK_BASE_URL ?? 'https://spark-api-open.xf-yun.com/x2').replace(/\/+$/, '')
const SPARK_MODEL = process.env.SPARK_MODEL ?? 'spark-x'
const SPARK_API_PASSWORD = (process.env.SPARK_API_PASSWORD ?? '').trim()

// 创建axios实例，添加网络配置
const axiosInstance = axios.create({
  timeout: 30000,
  httpAgent: new http.Agent({
    keepAlive: true,
    family: 4 // 强制使用IPv4，避免IPv6解析问题
  }),
  httpsAgent: new https.Agent({
    keepAlive: true,
    family: 4 // 强制使用IPv4，避免IPv6解析问题
  })
})

export const createSparkStream = async ({
  messages,
  model = SPARK_MODEL,
  temperature = 0.7,
  max_tokens = 2000,
}) => {
  if (!SPARK_API_PASSWORD) {
    throw new Error('Missing SPARK_API_PASSWORD in server environment.')
  }

  try {
    console.log('Attempting to connect to:', SPARK_BASE_URL)
    const response = await axiosInstance.post(`${SPARK_BASE_URL}/chat/completions`, {
      model,
      messages,
      stream: true,
      temperature,
      max_tokens,
    }, {
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${SPARK_API_PASSWORD}`,
      },
      responseType: 'stream'
    })

    return response
  } catch (error) {
    console.error('Error connecting to Spark API:', error)
    if (error.code === 'ENOTFOUND') {
      throw new Error('无法解析讯飞星火API域名，请检查网络连接和DNS配置')
    }
    throw error
  }
}

File: backend/pyserver/app/spark_client.py
from typing import Any, AsyncIterator

import httpx

from app import config


async def iter_spark_chat_stream(
    *,
    messages: list[dict[str, Any]],
    model: str | None,
    temperature: float,
    max_tokens: int,
) -> AsyncIterator[bytes]:
    if not config.SPARK_API_PASSWORD:
        raise ValueError("Missing SPARK_API_PASSWORD in server environment.")

    body = {
        "model": model or config.SPARK_MODEL,
        "messages": messages,
        "stream": True,
        "temperature": temperature,
        "max_tokens": max_tokens,
    }
    url = f"{config.SPARK_BASE_URL}/chat/completions"
    headers = {
        "Content-Type": "application/json",
        "Authorization": f"Bearer {config.SPARK_API_PASSWORD}",
    }

    timeout = httpx.Timeout(60.0, connect=30.0)
    async with httpx.AsyncClient(timeout=timeout) as client:
        async with client.stream("POST", url, json=body, headers=headers) as response:
            if response.status_code >= 400:
                err_text = (await response.aread()).decode("utf-8", errors="replace")
                raise RuntimeError(f"Spark API {response.status_code}: {err_text}")
            async for chunk in response.aiter_bytes():
                if chunk:
                    yield chunk
